sdcm/cluster_k8s/__init__.py

```diff
@@ -1748,13 +1748,17 @@ def hard_reboot(self):
             namespace=self.parent_cluster.namespace,
             timeout=self.pod_terminate_timeout * 60 + 10)
 
+        self.refresh_ip_address()
```

Comment:
I think this place affected by the race condition.
This IP refresh may be called when expected pod may not yet be created.

Reply:
you have better suggestion, of when we should do that ? maybe `_wait_public_ip` ?

Reply:
or maybe refactoring `Nemsis._kubernetes_wait_till_node_up_after_been_recreated` into the node ?

do we need to reconfigured the monitor ? is the monitor is deployed on the k8s cluster ? isn't that done automatically ?

Reply:
Just waiting for the pod creation (<5sec) and IP appearance (<15sec) is enough.
Monitoring is the common one - VM-based.
Yes, running reconfigure of a monitor is needed.

AS a result: as we can see, you are fixing here 2 different problems which can be in separate PRs.